projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -168,6 +191,23 @@ export class ExplorerComponent {
       this.initialState$.pipe(map(value => value.contextToggle.value.dashboardContext)),
       this.contextChangeSubject
     );
+    this.featureStateResolver.getFeatureState(ApplicationFeature.SavedQueries).subscribe(featureState => {
+      this.enableSavedQueries = featureState === FeatureState.Enabled ? true : false;
+    });
+  }
+
+  public onClickSaveQuery(): void {
+    this.preferenceService.get('savedQueries', []).subscribe(queries => {
+      this.savedQueries = queries as SavedQuery[];
+    });
+
+    const currentScope =
+      this.filters[0]?.metadata.scope === SPAN_SCOPE ? ScopeQueryParam.Spans : ScopeQueryParam.EndpointTraces;
+    const currentFilters = this.filters.map(filter => filter.urlString);
+
+    this.savedQueries.push({ scope: currentScope, filters: currentFilters });
```

Comment:
Sorry for checking this and telling about this now, but found something called `buildNavParamsWithFilters` in `ExplorerService`. I think that would be much simpler to use when we want to apply query. Currently, it saves the whole filter with all the metadata rather than just the `urlString`. Can you have a look at it once? I think it might be easier to save the whole filter in the storage and then call upon this method with `scope` and respective filter array.

Or we can add another method to the same service which takes an array of `filterUrlString` and `scope` and navigates to explorer.

Open for discussion around the same.

Reply:
Need to check after using the `buildNavParamsWithFilters` method and seeing if it works well.

Will address this in: https://github.com/razorpay/hypertrace-ui/pull/35
Have left a review comment for the same.